buildconfig/Jenkins/conda-setup

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# This script expects a environment variable called $WORKSPACE to be set.
+
+# From https://stackoverflow.com/a/9107028 by https://stackoverflow.com/users/1184238/andrew-norrie
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+EXPECTED_MAMBAFORGE_PATH=$HOME/mambaforge
```

Comment:
I thought we were going to avoid `$HOME` because:
* it avoids cross-job contamination
* enables testing different versions of Mamaforge/Miniconda without updating a whole machine installation.
* if it's inside the job workspace then a clean build will do what we want automatically and do everything from the start.

Reply:
I agree having some CI going as we develop the conda setup further is important. I think we need to make the change to the installation location before any of the pull request builds start using it.
One thing that would be useful in the interim I think is to have an option on the Jenkins job to delete the mambaforge installation - just in case it needs removing to try a full clean setup of that job.